SIPS/sip-83.md

```diff
@@ -47,55 +71,111 @@ allows subsequent minting and burning transactions to rely on the calculated deb
+The rationale of the debt pool snapshot value is the ability to capture a snapshot of all the synths prices and their
+respective synth's totalSupply (collectively as the `totalIssuedSynths`) for the purposes of minting, burning and
+claiming rewards.
 
-The snapshot is updated when minting and burning sUSD synths as sUSD synths don't require a new price update but is a basic addition / subtraction to the last snapshot value.
+The snapshot is updated when minting and burning sUSD synths as sUSD synths don't require a new price update but is a
+basic addition / subtraction to the last snapshot value.
 
-Having a public function that reads the latest prices and synth total supply on-chain and writes the new debt pool snapshot also provides a decentralised mechanims for the snapshot value to be executed on chain. Initially there will be no incentives to run this transaction to update the snapshot, future iterations of the debt pool snapshot update could rely on funds contributed by SNX stakers on each minting and burning transaction as there is a very high utility for them in gas savings.
+Having a public function that reads the latest prices and synth total supply on-chain and writes the new debt pool
+snapshot also provides a decentralised mechanism for the snapshot value to be executed on chain.
+Initially there will be no incentives to run this transaction to update the snapshot, future iterations of the debt pool
+snapshot update could rely on funds contributed by SNX stakers on each minting and burning transaction as there is a
+very high utility for them in gas savings.
 
 ### Technical Specification
 
 <!--The technical specification should outline the public API of the changes proposed. That is, changes to any of the interfaces Synthetix currently exposes or the creations of new ones.-->
 
-- `uint debtPoolSnapshot` Snapshot value of all the circulating `synths supply * synth prices` (in \$USD) at the last snapshot. Includes any extra sUSD minted or burned since the last snapshot as these are added directly to the total Issued Synths value and debt pool.
+- `uint debtPoolSnapshot` Snapshot value of all the circulating `synths supply * synth prices` (in \$USD) at the last snapshot. Includes any extra sUSD minted or burned since the last snapshot as these are added directly to the total Issued Synths value and debt pool. Wherever `Issuer.totalIssuedSynths()` is currently used, the snapshot will be substituted instead.
 
-- `uint snapshotTimestamp` Timestamp of the debt pool snapshot: Updated when `Issuer.updateDebtPoolSnapshot()` is executed but not updated when minting and burning transactions udpate the snapshot value.
+- `uint snapshotTimestamp` Timestamp of the debt pool snapshot: Updated when `Issuer.updateDebtPoolSnapshot()` is executed but not updated when mint, burn, or exchange transactions update the snapshot value.
```

Comment:
as above, we need the invalidity flag as well so to prevent issuance actions in the case of invalid rates.

Reply:
Updated.